internal/dag/httpproxy_processor.go

```diff
@@ -120,6 +121,13 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 			"Spec.VirtualHost.Fqdn must be specified")
 		return
 	}
+
+	// Check if FQDN is conformant according to RFC1123
+	if errs := validation.IsDNS1123Label(host); len(errs) != 0 {
```

Comment:
The reason you're getting a lot of test failures here is that a DNS label is a single *component* of a hostname.

So, in `subdomain.foo.com`, `subdomain`, `foo`, and `com` are all labels. A period (`.`) is not permitted in a DNS label.

I'd suggest using `IsFullyQualifiedDomainName` instead, however, if it fails, you also need to check for a single DNS label (which we accept as valid).

So, something like:
```suggestion
	if errs := validation.IsFullyQualifiedDomainName(host); len(errs) != 0 {
	  if labelErr := validation.IsDNS1123Label; labelErr != nil {
	    // add the error
```
I think something like that should help clear up your errors, and then you just need to make sure that you're testing both some fully qualified names (like `subdomain.foo.com`, `foo.com`), and a single label (like `foo`).

Reply:
Thanks @youngnick . FIxed in  latest patch.